### PR TITLE
feature/TECH 300 tag deployments

### PR DESCRIPTION
**Pull request by koeves**
Monorepo counterpart: https://github.com/Vandebron/Vandebron/pull/10482

- Get changes in merge commit
- Add github action to create tag
- Jenkins logs are printed stdout to stdout with `--follow`

[TECH-300]: https://vandebron.atlassian.net/browse/TECH-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TECH-300]: https://vandebron.atlassian.net/browse/TECH-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/mpyl/utilities/repo/__init__.py [hunk 1]
```diff
@@ -73,14 +73,21 @@ def latest_tag(self) -> str:
...
         in_branch = self.changes_in_branch()
         in_branch.append(Revision(len(in_branch), self.get_sha, self.changes_in_commit()))
         return in_branch
 
+    def changes_in_merge_commit(self) -> list[Revision]:
+        curr_rev = self._repo.rev_parse('HEAD')
+        parent_revs = reversed(list(curr_rev.iter_parents(n=1)))
+        return [Revision(count, str(rev),
+                         self._repo.git.diff_tree(no_commit_id=True, name_only=True, r=str(rev)).splitlines())
+                for count, rev in enumerate(parent_revs)]
```
Comment (SamTheisens) [inline src/mpyl/utilities/repo/__init__.py:89]: We assume that `HEAD` points to a tagged revision. Perhaps you could add a verification to make sure that you're indeed looking at the expected tag?
In that case you could also throw a clear error if we're not looking at a merge commit.

File: src/mpyl/utilities/jenkins/runner.py [hunk 2]
```diff
@@ -159,4 +160,5 @@ def run(self):
 
         new_build_number = last_build_number + 1
 
-        self.follow_logs(job, new_build_number, last_build)
+        if self.follow:
+            self.follow_logs(job, new_build_number, last_build)
```
Comment (SamTheisens) [inline src/mpyl/utilities/jenkins/runner.py:164]: Perhaps there's value in a `silent mode` too? Where you block (showing progress bar) until the build is done, but don't stream the logs.

Not for this PR, but just a thought.

File: src/mpyl/cli/commands/build/mpyl.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ class MpylRunConfig:
 @dataclass(frozen=True)
 class MpylCliParameters:
     local: bool
+    tag: bool = False
```
Comment (SamTheisens) [inline src/mpyl/cli/commands/build/mpyl.py:32]: Perhaps the tag could be an `Optional[str]` to validate (see later comment)

File: mpyl-example.py [hunk 4]
```diff
@@ -78,6 +78,7 @@ def main(log: Logger, args: argparse.Namespace):
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Simple MPL pipeline')
     parser.add_argument('--local', '-l', help='a local developer run', default=False, action='store_true')
+    parser.add_argument('--tag', '-t', help='A tag run to deploy to test/acce/prod', default=False, action='store_true')
```
Comment (SamTheisens) [inline mpyl-example.py:81]: I think we probably need a target (test/acc/prod) parameter too. Not for this PR, but we will need it I think.

**Review by SamTheisens — COMMENTED** [review]
LGTM in general. Some additional error handling would be good I think. I'm afraid this will be rather puzzling to people if things go wrong and the intent isn't clear.

**Review by SamTheisens — COMMENTED** [review]
LGTM in general. Some additional error handling would be good I think. I'm afraid this will be rather puzzling to people if things go wrong and the intent isn't clear.

File: mpyl-example.py (hunk 4 above)
Comment (koeves) [reply mpyl-example.py:81]: It's handled by an environment variable in the run properties. Check also https://github.com/Vandebron/Vandebron/pull/10482

File: src/mpyl/utilities/jenkins/runner.py (hunk 2 above)
Comment (koeves) [reply src/mpyl/utilities/jenkins/runner.py:164]: I was doubting between the two approaches, if it should be silent or follow by default. But your version seems like a nice idea as well. Maybe let's hear what the users prefer as the default

File: src/mpyl/utilities/repo/__init__.py (hunk 1 above)
Comment (koeves) [reply src/mpyl/utilities/repo/__init__.py]: My current thinking is that I can get the tag associated with the commit by doing `git describe --tags curr_rev`. This should match the $TAG_NAME in jenkins that we pass to mpyl by `--tag $TAG_NAME`. So indeed what you suggested, the tag will be a parameter not a flag

**Review by SamTheisens — APPROVED** [review]

**Pull request merged**